sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/query/DocumentQueryExecutionContextFactory.java

```diff
@@ -78,49 +84,105 @@
         }
 
         Instant startTime = Instant.now();
-        Mono<PartitionedQueryExecutionInfo> queryExecutionInfoMono =
-            QueryPlanRetriever
-                .getQueryPlanThroughGatewayAsync(diagnosticsClientContext, client, query, resourceLink);
+        Mono<PartitionedQueryExecutionInfo> queryExecutionInfoMono;
+        if (queryPlanCachingEnabled &&
+                isScopedToSinglePartition(cosmosQueryRequestOptions) &&
+                queryPlanCache.containsKey(query.getQueryText())) {
```

Comment:
getQueryText will return the query as specified by the user - so if user isn't using query parameters we would see a very high number of different queries even for the same shape, correct? I though the Sql parsing done with ANTLR would allow us to create the normalized query shape - like take parameter/filter values out of the equation etc.?
I might be missing some context - so more a question that an ask to change anything...

Reply:
Yes thats true, it would result in high number, thats why we want to cap the size of the cache. Even if we parse the query using ANTLR, we wouldn't build any context on the query info at this point and just classify the query as locally executable or not